centreon/src/Core/Media/Infrastructure/API/UpdateMedia/UpdateMediaController.php

```diff
@@ -0,0 +1,91 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types = 1);
+
+namespace Core\Media\Infrastructure\API\UpdateMedia;
+
+use Centreon\Application\Controller\AbstractController;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Common\Infrastructure\Upload\FileCollection;
+use Core\Media\Application\UseCase\AddMedia\AddMediaRequest;
+use Core\Media\Application\UseCase\UpdateMedia\UpdateMedia;
+use Core\Media\Infrastructure\API\AddMedia\AddMediaValidator;
+use Core\Media\Infrastructure\API\Exception\MediaException;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+final class UpdateMediaController extends AbstractController
+{
+    use LoggerTrait;
+
+    public function __invoke(Request $request, UpdateMedia $useCase, UpdateMediaPresenter $presenter): Response
+    {
+        $this->denyAccessUnlessGrantedForApiConfiguration();
+        $uploadedFile = '';
+        $filesToDeleteAfterProcessing = [];
+        try {
+            $assertion = new AddMediaValidator($request);
+            $assertion->assertFilesSent();
+            $assertion->assertDirectory();
+
+            $fileIterator = new FileCollection();
+
+            /** @var UploadedFile|list<UploadedFile> $files */
+            $files = $request->files->get('data');
+            if (is_array($files)) {
+                foreach ($files as $file) {
+                    $fileIterator->addFile($file);
+                    $filesToDeleteAfterProcessing[] = $file->getPathname();
+                }
+            } else {
+                $fileIterator->addFile($files);
+                $filesToDeleteAfterProcessing[] = $files->getPathname();
+            }
+            $updateMediaRequest = new AddMediaRequest($fileIterator->getFiles());
+            $updateMediaRequest->directory = (string) $request->request->get('directory');
+
+            $useCase($updateMediaRequest, $presenter);
+        } catch (MediaException $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+            $presenter->presentResponse(new ErrorResponse($ex->getMessage()));
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+            $presenter->presentResponse(
+                new ErrorResponse(
+                    MediaException::errorUploadingFile($uploadedFile)->getMessage()
+                )
+            );
+        } finally {
+            foreach ($filesToDeleteAfterProcessing as $fileToDelete) {
+                if (is_file($fileToDelete)) {
+                    unlink($fileToDelete);
+                    $this->debug('Deleting the uploaded file', ['filename' => $fileToDelete]);
```

Comment:
debug level is only used for the debugging note for production code

Reply:
still present :) the level should be info because debug it's only for a code in development :)